### PR TITLE
Dev handler

### PR DESCRIPTION
**Pull request by reszczyk**
Added exception handling for the following errors:

- Model not found
- Not found http
- Method not allowed
- Token mismatch

### CODE REVIEW

File: app/Exceptions/Handler.php [hunk 1]
```diff
@@ -1,19 +1,22 @@
...
 
 class Handler extends ExceptionHandler
 {
+    use HandlerExceptions;
     /**
      * A list of the exception types that are not reported.
      *
      * @var array
      */
     protected $dontReport = [
```
Comment (krzysztofrewak) [inline app/Exceptions/Handler.php]: I think `$dontReport` is empty in `Illuminate\Foundation\Exceptions\Handler`, so you can remove it from here.

File: app/Exceptions/Handler.php [hunk 2]
```diff
@@ -29,27 +32,24 @@ class Handler extends ExceptionHandler
     /**
      * Report or log an exception.
      *
-     * @param  \Throwable  $exception
-     * @return void
-     *
      * @throws \Exception
      */
-    public function report(Throwable $exception)
+    public function report(Throwable $exception): void
```
Comment (krzysztofrewak) [inline app/Exceptions/Handler.php]: Do we need this empty method here?

File: app/Exceptions/Handler.php [hunk 3]
```diff
@@ -29,27 +32,24 @@ class Handler extends ExceptionHandler
...
-     * @param  \Illuminate\Http\Request  $request
-     * @param  \Throwable  $exception
+     * @param \Illuminate\Http\Request $request
+     *
      * @return \Symfony\Component\HttpFoundation\Response
      *
      * @throws \Throwable
      */
-    public function render($request, Throwable $exception)
+    public function render($request, Throwable $exception): response
```
Comment (krzysztofrewak) [inline app/Exceptions/Handler.php]: There will be a definitely problem with this small `r` in `response` type return on unix systems. ;)

File: app/Exceptions/HandlerExceptions.php [hunk 4]
```diff
@@ -0,0 +1,36 @@
...
+use Illuminate\Database\Eloquent\ModelNotFoundException;
+use Illuminate\Http\JsonResponse;
+use Illuminate\Http\Response;
+use Illuminate\Session\TokenMismatchException;
+use Symfony\Component\HttpKernel\Exception\MethodNotAllowedHttpException;
+use Symfony\Component\HttpKernel\Exception\NotFoundHttpException;
+
+trait HandlerExceptions
+{
+    public function apiException($request, $exception): JsonResponse
```
Comment (krzysztofrewak) [inline app/Exceptions/HandlerExceptions.php]: Add type hints.

File: routes/api.php [hunk 5]
```diff
@@ -27,12 +27,10 @@
     echo 'OpenTransportManager API';
 });
 
-Route::fallback(function (): RedirectResponse {
+Route::get('/docs', function (): RedirectResponse {
```
Comment (krzysztofrewak) [inline routes/api.php]: Maybe it would be a good idea to not litter your routing and move this redirect into `DocsControllers` or something?

**Review by krzysztofrewak — CHANGES_REQUESTED** [review]

**Review by krzysztofrewak — APPROVED** [review]

File: app/Exceptions/Handler.php (hunk 2 above)
Comment (reszczyk) [reply app/Exceptions/Handler.php]: We could use it for logger which is not implemented yet. Even if we implement logger, report method isn't necessary to log exceptions from handler. When we decide to use it, I'll just restore it.

**Pull request merged**